tap_redshift/__init__.py

```diff
@@ -366,35 +366,49 @@ def sync_table(connection, catalog_entry, state):
         elif replication_key is not None:
             select += ' ORDER BY {} ASC'.format(replication_key)
 
-        time_extracted = utils.now()
-        query_string = cursor.mogrify(select, params)
-        LOGGER.info('Running {}'.format(query_string))
-        cursor.execute(select, params)
-        row = cursor.fetchone()
-        rows_saved = 0
-
-        with metrics.record_counter(None) as counter:
-            counter.tags['database'] = catalog_entry.database
-            counter.tags['table'] = catalog_entry.table
-            while row:
-                counter.increment()
-                rows_saved += 1
-                record_message = row_to_record(catalog_entry,
-                                               stream_version,
-                                               row,
-                                               columns,
-                                               time_extracted)
-                yield record_message
-
-                if replication_key is not None:
-                    state = singer.write_bookmark(state,
-                                                  tap_stream_id,
-                                                  'replication_key_value',
-                                                  record_message.record[
-                                                      replication_key])
-                if rows_saved % 1000 == 0:
-                    yield singer.StateMessage(value=copy.deepcopy(state))
-                row = cursor.fetchone()
+        if limit > 0:
+            select += ' LIMIT %(limit)s OFFSET %(offset)s '
+            params['limit'] = limit
+            params['offset'] = 0
+
+        while True:
```

Comment:
```suggestion
        while more_records:
```
Re-write without `break`.